wsdl-cli/src/main/java/io/ballerina/wsdl/cli/WsdlCmd.java

```diff
@@ -0,0 +1,159 @@
+/*
+ *  Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com)
+ *
+ *  WSO2 LLC. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing,
+ *  software distributed under the License is distributed on an
+ *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ *  KIND, either express or implied.  See the License for the
+ *  specific language governing permissions and limitations
+ *  under the License.
+ */
+
+package io.ballerina.wsdl.cli;
+
+import io.ballerina.cli.BLauncherCmd;
+import io.ballerina.wsdl.core.WsdlToBallerina;
+import io.ballerina.wsdl.core.WsdlToBallerinaResponse;
+import io.ballerina.wsdl.core.generator.GeneratedSource;
+import picocli.CommandLine;
+
+import java.io.File;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintStream;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import static io.ballerina.wsdl.cli.Messages.MISSING_WSDL_PATH;
+
+/**
+ * Command-line utility class for generating Ballerina source code from WSDL files.
+ * It uses the specified WSDL path and optional operation names to generate the code.
+ *
+ * @since 0.1.0
+ */
+@CommandLine.Command(
+        name = "wsdl",
+        description = "Generate the Ballerina sources for a given WSDL definition."
+)
+public class WsdlCmd implements BLauncherCmd {
+    private static final String CMD_NAME = "wsdl";
+    private final PrintStream outStream;
+    private final boolean exitWhenFinish;
+
+    @CommandLine.Option(names = {"-i", "--input"}, description = "Relative path to the WSDL file")
+    private String inputPath;
+
+    @CommandLine.Option(names = {"--operations"}, description = "Comma-separated operation names to generate")
+    private String operations;
+
+    @CommandLine.Option(names = {"--help", "-h"}, hidden = true)
+    private boolean helpFlag;
+
+    public WsdlCmd() {
+        this.outStream = System.err;
+        this.exitWhenFinish = true;
+    }
+
+    @Override
+    public void execute() {
+        if (inputPath == null || inputPath.isBlank()) {
+            outStream.println(MISSING_WSDL_PATH);
+            exitOnError();
+            return;
+        }
+
+        List<String> operationList = new ArrayList<>();
+        if (operations != null && !operations.isBlank()) {
+            operationList.addAll(Arrays.asList(operations.split(",")));
+        }
+
+        try {
+            wsdlToBallerina(inputPath, operationList);
+        } catch (IOException e) {
+            outStream.println("Error: " + e.getLocalizedMessage());
```

Comment:
Wouldn't it be better to append something more specific than "Error:" here? Error while reading the wsdl file, error while writing Ballerina records etc?